FirebaseSessions/Sources/FirebaseSessions.swift

```diff
@@ -26,13 +28,42 @@ protocol SessionsProvider {
 @objc(FIRSessions) class Sessions: NSObject, Library, SessionsProvider {
   // MARK: - Private Variables
 
-  /// The app associated with all sessions.
-  private let googleAppID: String
+  /// The Firebase App ID associated with Sessions.
+  private let appID: String
+
+  /// Top-level Classes in the Sessions SDK
+  private let main: Main
+  private let initiator: Initiator
+  private let identifiers: Identifiers
 
   // MARK: - Initializers
 
-  required init(app: FirebaseApp) {
-    googleAppID = app.options.googleAppID
+  required convenience init(app: FirebaseApp) {
```

Comment:
I like the idea of reducing our dependency surface to just the functions we need. One thing I've done instead of passing a function is kept the protocol in there because we'll be using that pattern all over the SDK and I am trying to reuse patterns as much as possible.

Reply:
Discussed offline. Will hold off.